### PR TITLE
Make preserve data in the browser feature

### PR DESCRIPTION
**Pull request by hienphan0111**
## Create preserve data in the browser feature 

implement data preservation in the browser according to the following requirements:

- [ ] When the user changes the content of any input field, the data is saved to the local storage.

- [ ] When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

### CODE REVIEW

**Review by jerryowusu — COMMENTED** [review]
Hi Team, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

invalid

## Invalid Code Review Request

You have submitted a pair programming project which was done by only one person. Kindly make sure that both members have made a commit to the project to indicate participation :wink:

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: index.js [hunk 1]
```diff
@@ -172,3 +174,46 @@ form.addEventListener('submit', (event) => {
...
+  storeData(email, 'email');
+});
+
+message.addEventListener('change', () => {
+  storeData(message, 'message');
+});
+
+document.addEventListener('DOMContentLoaded', () => {
+  contactFormData = JSON.parse(localStorage.getItem('contactFormData'));
+  if (contactFormData.data.length > 0) {
```
Comment (ha-manel) [inline index.js]: - [ ] The data preservation is not working as you can see below 👇 Kindly fix the issues 👍 

Animation

This is local storage, it's not saving either
image

And I have this error in console about this line (#214) 👇 
image

**Review by ha-manel — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES ♻️

## Hi Team👋 
Great job so far, There are some changes that need to be implemented in order to proceed to the next project 👍  



## Highlights:  
* All tests passed ✔️
* Meaningful commit messages ✔️ 
* Professional README file ✔️

## Required Changes:

- [ ] @Ferizad Please consider using best practices when you write commit messages, they should be capitalized sentences in the imperative form, example: "Add linters" instead of "added linters" or "linters are added" ... etc 👍 

_Kindly check the comments below_


------
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by GedeonTS — APPROVED** [review]
# STATUS: APPROVED :tada: :clap: :champagne: 
Hi @hienphan0111 ,

You are amazing :smiley: :clap: !  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

 

## Highlights :clap: :tophat: 
- Data is getting stored in local storage :+1: 
- Good use of GitHub flow :+1: 
- Linter checks are passing
## Optional suggestions :small_airplane: 
- [OPTIONAL ] As suggested by the previous reviewer, please make sure you write more __professional_ commit messages, describing the changes you are making clearly in the imperative form.
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**